WebRoot/h/changepass

```diff
@@ -445,12 +492,52 @@
                 errorMessageDiv.style.display = "none";
             }
 
-            setSaveButtonDisabled(enabledRules.length > matchedRule.length || confirmPasswordInput.value !== newPasswordInput.value);
+            handleConfirmPasswordChange();
         };
 
         function handleConfirmPasswordChange() {
             resetImg(confirmPasswordInput.value === newPasswordInput.value, getElement("mustMatchCheckImg"), getElement("mustMatchCloseImg"));
-            setSaveButtonDisabled(confirmPasswordInput.value !== newPasswordInput.value);
+            setSaveButtonDisabled();
+        };
+
+        function dryRun() {
+            matchedRemoteRules = [];
+            var request = {
+                Body: {
+                    ChangePasswordRequest:{
+                        account: {
+                            by : "name",
+                            _content : "${mailbox.accountInfo.name}"
+                        },
+                        oldPassword: getElement("oldPassword").value,
+                        password: newPasswordInput.value,
+                        _jsns: "urn:zimbraAccount",
+                        dryRun: "1"
+                    }
+                },
+                Header: {
+                    context: {
+                        _jsns: "urn:zimbra",
+                        csrfToken: "${mailbox.csrfToken}"
+                    }
+                }
+            };
+            var xmlhttp=new XMLHttpRequest();
+            xmlhttp.open("POST","/service/soap/ChangePasswordRequest",false);
+            xmlhttp.send(JSON.stringify(request));
+
+            if (xmlhttp.status == 200 || xmlhttp.status == 201) {
+                var jsonResponse = JSON.parse(xmlhttp.responseText);
+                var changePasswordResponse = jsonResponse.Body.ChangePasswordResponse;
+                matchedRemoteRules.push({type : "zimbraPasswordEnforceHistory"});
+                compareMatchedRemoteRules(matchedRemoteRules);
+            } else {
+                var jsonResponse = JSON.parse(xmlhttp.responseText);
+                var errorDetails = jsonResponse.Body.Fault.Detail.Error;
+                if (errorDetails.Code == "account.PASSWORD_RECENTLY_USED") {
```

Comment:
what about common passwords check?

Reply:
It is not yet implemented in Z9 environment.
Back end team will require more time and major release to provide zimbraPasswordBlockCommonEnabled in GetInfoResponse.
Refer : https://jira.corp.synacor.com/browse/ZCS-9767

Reply:
can we complete it from UI side for now? when backend will be fixed everything will automatically start working instead of waiting for client side changes then

Reply:
Fixed.